mesa/model.py

```diff
@@ -185,10 +185,23 @@ def run_model(self) -> None:
     def step(self) -> None:
         """A single step. Fill in here."""
 
-    def next_id(self) -> int:
-        """Return the next unique ID for agents, increment current_id"""
+    @property
+    def _next_id(self) -> int:
+        """Return the next unique ID for agents. Starts at 0."""
+        return_id = self.current_id
         self.current_id += 1
-        return self.current_id
+        return return_id
+
```

Comment:
in my opinion, this is part of the behavior of the Agent class so it is strange that it is located in the model class.

Reply:
Agreed, but let's move it in a separate PR. It breaks practically all tests with more than one test class or function in a file.

Reply:
Not sure I fully understand your point. Why not move the counting code into Agent in this PR? 

I haven't looked at how bad this would be in terms of tests, but it seems logical to include it here?

Reply:
Copy my branch and move it. You will see what happens, and maybe you have an elegant solution for it.